actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -72,8 +107,77 @@ module TagHelper
       #
       #   tag("div", data: {name: 'Stephen', city_state: %w(Chicago IL)})
       #   # => <div data-name="Stephen" data-city-state="[&quot;Chicago&quot;,&quot;IL&quot;]" />
-      def tag(name, options = nil, open = false, escape = true)
-        "<#{name}#{tag_options(options, escape) if options}#{open ? ">" : " />"}".html_safe
+      # 
+      # === Modern syntax
+      # Modern syntax follows one of two formats:
+      #   tag.<name>(options, escape)
+      #   tag.<name>(content, options, escape)
+      # Returns an HTML tag. Content has to be a string. If content is passed
+      # than tag is surrounding the content. Otherwise tag will be empty. Youl
+      # can also use a block to pass the content inside ERB templates. Result
+      # is by default HTML5 compliant. Set escape parameter to false to
+      # disable attribute value escaping. The tag will be generated with
+      # related closing tag unless tag represents a
+      # void[https://www.w3.org/TR/html5/syntax.html#void-elements] element.
+      # Method will rise NoMethodError if element is not a part of the HTML5
+      # standard. It will rise ArgumentError if you try to pass content for a 
+      # void element.
+      #
+      # ==== Options
+      # Like with traditional syntax the options hash can be used with
+      # attributes with no value (like disabled and readonly), which you can
+      # give a value of true in the options hash. You can use symbols or
+      # strings for the attribute names.
+      #    
+      # ==== Examples
+      #   tag.span
+      #   # => <span></span>
+      #
+      #   tag.span(class: "bookmark")
+      #   # => <span class=\"bookmark\"></span>
+      #
+      #   tag.input type: 'text', disabled: true
+      #   # => <input type="text" disabled="disabled">
+      #
+      #   tag.input type: 'text', class: ["strong", "highlight"]
+      #   # => <input class="strong highlight" type="text">
+      #
+      #   tag.img src: "open & shut.png"
+      #   # => <img src="open &amp; shut.png">
+      #
+      #   tag.img({src: "open &amp; shut.png"}, false)
+      #   # => <img src="open &amp; shut.png">
+      #
+      #   tag.div(data: {name: 'Stephen', city_state: %w(Chicago IL)})
+      #   # => <div data-name="Stephen" data-city-state="[&quot;Chicago&quot;,&quot;IL&quot;]"></div>
+      #
+      #   tag.p "Hello world!"
+      #    # => <p>Hello world!</p>
```

Comment:
From here and below you switch to indenting output by one space. Think it would read nicer without that single space indent and be more consistent with the above examples.